src/TRestGeant4Metadata.cxx

```diff
@@ -778,8 +781,8 @@ void TRestGeant4Metadata::InitFromConfigFile() {
     string seedstr = GetParameter("seed", "0");
     if (ToUpper(seedstr) == "RANDOM" || ToUpper(seedstr) == "RAND" || ToUpper(seedstr) == "AUTO" ||
         seedstr == "0") {
-        double* dd = new double();
-        fSeed = (uintptr_t)dd + (uintptr_t) this;
+        auto* dd = new double();
```

Comment:
That's bizarre, there is no better way to do this?
If fSeed is the seed for the random initialization you can use time(null) or  std::random_device

Reply:
You mean getting the pointer value as an integer to be used as the seed? I think it is quite an original one, and I believe introduced by @nkx111 . If you call 100 times within a second to `time(null)` you will get the same value. Then, when launching batch jobs to a cluster it happens exactly that, it will launch all the jobs simultaneously and all jobs will get the same seed. That was a puzzling problem because we were getting 100 simulations back, which were repeated.

Reply:
Did you try std::random_device? You can also use std::chrono::high_resolution_clock in nanoseconds for the seed, it is very odd to have all the simulations starting in the same nanosecond.
Note that to get the pointer value although original, it is not guaranteed to be different every time since it should be inside the boundaries of the memory addresses. This might happen if you run the similation several times in a row.

Reply:
No, we didnt try `std::random_device` that solution was working. In the other hand, the seed is the result of the address of two objects, the local variable address (that I doubt it will get the same address as another program, must be low probability), and the memory address of the object itself (`this`) that must be different and it is allocated, therefore reserved, all along the execution of the program, i.e. usually during hours.

Reply:
```
root [1] TRestRun *r = new TRestRun()
(TRestRun *) 0x55b018003fd0
root [2] delete r
root [3] TRestRun *r = new TRestRun()
(TRestRun *) 0x55b0186e2be0
root [4] delete r
root [5] TRestRun *r = new TRestRun()
(TRestRun *) 0x55b0187a4e30
root [6] delete r
root [7] TRestRun *r = new TRestRun()
(TRestRun *) 0x55b0186c7f90
root [8] delete r
root [9] TRestRun *r = new TRestRun()
(TRestRun *) 0x55b0186cd9d0
```